modules/core/src/test/java/org/apache/ignite/internal/IgniteClientReconnectContinuousProcessorTest.java

```diff
@@ -305,6 +312,34 @@ public void testCacheContinuousQueryReconnectNewServer() throws Exception {
         }
     }
 
+    @Test
+    public void testCacheContinuousQuerySpliteratorMultipleCalls() {
+        Ignite client = grid(serverCount());
+
+        assertTrue(client.cluster().localNode().isClient());
+
+        IgniteCache<Object, Object> clientCache = client.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME));
+
+        CacheEventListener lsnr = new CacheEventListener();
+
+        ContinuousQuery<Object, Object> qry = new ContinuousQuery<>();
+
+        qry.setInitialQuery(new ScanQuery<>(new IgniteBiPredicate<Object, Object>() {
+            @Override public boolean apply(Object key, Object val) {
+                return key != null;
+            }
+        }));
+        qry.setAutoUnsubscribe(true);
+
+        qry.setLocalListener(lsnr);
+
+        try(QueryCursor<?> cur = clientCache.query(qry)) {
```

Comment:
Here you test the result of `ContinuousQuery` only, while other `Query` variations (e.g. `SqlQuery`, `SqlFieldsQuery`, `ScanQuery`) are available too and need to be tested.
See `org.apache.ignite.internal.processors.cache.IgniteCacheProxyImpl#query(...)` implementation(s).